WPMedia/fusion-news-theme-blocks
# PEN-1486: [CMG] Add ability to change Image position on small items in Top Table List

## Description
Modified small image promos to have 'imagePosition' custom field and decouple image position from 'storiesPerRow' setting. Also refactored and modified unit tests.

## Jira Ticket
- [PEN-1486](https://arcpublishing.atlassian.net/browse/PEN-1486)

## Acceptance Criteria
1. On Desktop, the setting for Stories Per Row currently changes the image position. It should be changed so that Stories Per Row does not affect Image Position. 
2. In the top Table List, a custom field Image position should be added under the Small Story Settings group, with the same options and behavior as described in  

## Test Steps
1. In `Fusion-News-Theme` repo, checkout `master` & `git pull`
2. After checking out this feature branch in `fusion-news-theme-blocks`, run `rm -rf ./node_modules && npx lerna clean && npm i && npx lerna bootstrap && cd blocks/header-nav-chain-block && npm i && cd ../..`
3. Run `npx fusion start -f -l @wpmedia/header-nav-chain-block` in `Fusion-News-Theme`
4. Open homepage in PB editor and add a `Top Table List` block to the main section
5. Click on the `Top Table List` to open in PB editor and use the following configuration:
![image](https://user-images.githubusercontent.com/26662906/99563477-eb04fe80-298e-11eb-8c10-0bcf37c97034.png)
6. Test all `Image Position` options (under "Small story settings") (`left`, `right`, `above`, `below`) with various combinations of `Stories per row`. Certain combinations such as putting the image right or left with 4 items per row will constrain the space quite a bit but the items should all still "fit" within the container (not "overflowing" the container or onto each other). Try to test all `Stories per row` options to make sure all layout combinations look good.

NOTE: I personally have been having trouble with my image resizer locally so the images are appearing broken. I'm not sure if this will be the case for whoever reviews but just a heads-up. My understanding is that this is an SDK issue with the `Image` component that is being fixed by the platform team. I did my testing using an "unresized" `img` tag and just reverted the code back to using the `Image` component before committing so everything _should_ work as expected once the image resizer issue is fixed.


## Effect Of Changes
### Before
The "image position" for small promo items in the `Top Table List` was tied to the `Stories per row` custom field. There was no way to customize the position of the image directly.

### After
There is now a separate `Image Position` custom field that is decoupled from the `Stories per row` configuration. You can now customize the positioning of the image independent of how many items you decide should render per row.

## Dependencies or Side Effects
_Examples of dependencies or side effects are:_
- Additional settings that will be required in the blocks.json
- Changes to the custom fields which will require users to reconfigure features
- Update to css framework or SDK
- Dependency on another PR that needs to be merged first

## Review Checklist
_The author of the PR should fill out the following sections to ensure this PR is ready for review._
please explain why (so that we can fix it whenever it gets refactored).
